platform/darwin/src/NSDictionary+MGLAdditions.mm

```diff
@@ -0,0 +1,51 @@
+#import "NSDictionary+MGLAdditions.h"
+
+#import "NSExpression+MGLAdditions.mm"
+
+@implementation NSDictionary (MGLAdditions)
+
+- (mbgl::PropertyMap)mbgl_propertyMap {
+    mbgl::PropertyMap propertyMap;
+    return [self propertyMapWithPropertyMap:propertyMap dictionary:self];
+}
+
+- (mbgl::PropertyMap)propertyMapWithPropertyMap:(mbgl::PropertyMap &)propertyMap dictionary:(NSDictionary *)dictionary {
+    for (NSString *key in dictionary.allKeys) {
+        if ([dictionary[key] isKindOfClass:[NSDictionary class]]) {
+            std::unordered_map<std::string, mbgl::Value> unorderedMap;
+            [self propertyMapWithPropertyMap:unorderedMap dictionary:dictionary[key]];
+            propertyMap[[key UTF8String]] = unorderedMap;
+        } else if ([dictionary[key] isKindOfClass:[NSArray class]]) {
+            NSArray *array = dictionary[key];
+            std::vector<mbgl::Value> vector;
+            propertyMap[[key UTF8String]] = [self vectorForNSArray:array vector:vector];
+        } else {
+            NSExpression *expression = [NSExpression expressionForConstantValue:dictionary[key]];
+            propertyMap[[key UTF8String]] = [expression mgl_filterValue];
+        }
+    }
+    
+    return propertyMap;
+}
+
+- (std::vector<mbgl::Value>)vectorForNSArray:(NSArray *)array vector:(std::vector<mbgl::Value> &)vector {
```

Comment:
Similarly, this would be a category method on NSArray that returns a vector of values, and the method name would be something like mgl_valueVector.